force-app/main/default/classes/GiftBatchForQueueable_TEST.cls

```diff
@@ -0,0 +1,439 @@
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2022
+* @description GiftBatchForQueueable test class
+*/
+@IsTest
+private class GiftBatchForQueueable_TEST {
+
+    @isTest
+    static void shouldAttemptToCaptureElevatePaymentsWhenServiceUnavailable() {
+        // Arrange
+        Id dummyDataImportBatchId = UTIL_UnitTestData_TEST.mockId(DataImportBatch__c.SObjectType);
+        GiftBatchForQueueable dummyGiftBatchForQueueable = buildGiftBatch(dummyDataImportBatchId);
+        List<DataImport__c> authorizedGift = new List<DataImport__c> {
+            new DataImport__c(
+                Id = UTIL_UnitTestData_TEST.mockId(DataImport__c.SObjectType),
+                Payment_Status__c = ElevateTransactionStatus.AUTHORIZED.name(),
+                Status__c = BDI_DataImport_API.bdiDryRunValidated
+            )
+        };
+
+        GiftsMock giftsMock = new GiftsMock().with(authorizedGift);
+        dummyGiftBatchForQueueable.gifts = stubFor(giftsMock);
+
+        ElevateBatchServiceMock elevateBatchServiceMock = new ElevateBatchServiceMock();
+        dummyGiftBatchForQueueable.elevateBatchService = stubFor(elevateBatchServiceMock);
+
+        PS_IntegrationServiceConfigMock integrationServiceConfigMock = new PS_IntegrationServiceConfigMock(true);
+        dummyGiftBatchForQueueable.paymentIntegrationServiceConfig = stubFor(integrationServiceConfigMock);
+
+        // Act
+        dummyGiftBatchForQueueable.captureElevateBatches();
+
+        // Assert
+        System.assertEquals(true, elevateBatchServiceMock.enqueuePaymentCaptureJobForCalled);
+        System.assertEquals(true, giftsMock.authorizedToPaymentProcessingStatusWasCalled);
+    }
+
+    @isTest
+    static void shouldNotAttemptToCaptureElevatePaymentsWhenServiceAvailable() {
```

Comment:
Are the names of this method and the first one in this test class switched? Looks like this one is when service is unavailable and the first one is when the service is available.